util/dnsname/dnsname.go

```diff
@@ -65,6 +60,17 @@ func ToFQDN(s string) (FQDN, error) {
 	return FQDN(raw), nil
 }
 
+func ToFQDN(s string) (FQDN, error) {
```

Comment:
Add a documentation string here that expresses that this explicitly strips a leading '.' and thereby prevents the creation of suffix matches.